api/index.js

```diff
@@ -5,409 +5,99 @@
-      }
-      return this[kRollups];
-    },
-  },
-  scripts_painless_execute: {
-    get() {
-      return this.scriptsPainlessExecute;
-    },
-  },
-  search_shards: {
-    get() {
-      return this.searchShards;
-    },
-  },
-  search_template: {
-    get() {
-      return this.searchTemplate;
-    },
-  },
-  security: {
-    get() {
-      if (this[kSecurity] === null) {
-        this[kSecurity] = new SecurityApi(this.transport, this[kConfigurationError]);
-      }
-      return this[kSecurity];
-    },
-  },
-  shutdown: {
-    get() {
-      if (this[kShutdown] === null) {
-        this[kShutdown] = new ShutdownApi(this.transport, this[kConfigurationError]);
-      }
-      return this[kShutdown];
-    },
-  },
-  snapshot: {
-    get() {
-      if (this[kSnapshot] === null) {
-        this[kSnapshot] = new SnapshotApi(this.transport, this[kConfigurationError]);
-      }
-      return this[kSnapshot];
-    },
-  },
-  tasks: {
-    get() {
-      if (this[kTasks] === null) {
-        this[kTasks] = new TasksApi(this.transport, this[kConfigurationError]);
-      }
-      return this[kTasks];
-    },
-  },
-  terms_enum: {
-    get() {
-      return this.termsEnum;
-    },
-  },
-  transforms: {
-    get() {
-      if (this[kTransforms] === null) {
-        this[kTransforms] = new TransformsAPi(this.transport, this[kConfigurationError]);
-      }
-      return this[kTransforms];
-    },
-  },
-  update_by_query: {
-    get() {
-      return this.updateByQuery;
-    },
-  },
-  update_by_query_rethrottle: {
-    get() {
-      return this.updateByQueryRethrottle;
-    },
-  },
-});
+  cat: { get() { return this[kApiLoader]('./cat/_api') } },
+  cluster: { get() { return this[kApiLoader]('./cluster/_api') } },
+  danglingIndices: { get() { return this[kApiLoader]('./dangling_indices/_api') } },
+  http: { get() { return this[kApiLoader]('./http/_api') } },
+  indices: { get() { return this[kApiLoader]('./indices/_api') } },
+  ingest: { get() { return this[kApiLoader]('./ingest/_api') } },
+  knn: { get() { return this[kApiLoader]('./knn/_api') } },
+  ml: { get() { return this[kApiLoader]('./ml/_api') } },
+  nodes: { get() { return this[kApiLoader]('./nodes/_api') } },
+  notifications: { get() { return this[kApiLoader]('./notifications/_api') } },
+  ppl: { get() { return this[kApiLoader]('./ppl/_api') } },
+  remoteStore: { get() { return this[kApiLoader]('./remote_store/_api') } },
+  rollups: { get() { return this[kApiLoader]('./rollups/_api') } },
+  searchPipeline: { get() { return this[kApiLoader]('./search_pipeline/_api') } },
+  security: { get() { return this[kApiLoader]('./security/_api') } },
+  snapshot: { get() { return this[kApiLoader]('./snapshot/_api') } },
+  sql: { get() { return this[kApiLoader]('./sql/_api') } },
+  tasks: { get() { return this[kApiLoader]('./tasks/_api') } },
+  transforms: { get() { return this[kApiLoader]('./transforms/_api') } },
+})
```

Comment:
It probably makes sense to have an explicit `loadAll()` as well to avoid warmup time during service. Doesn't need to be in this PR.